File: lib/shared/widgets/custom_button.dart
import 'package:flutter/material.dart';
import '../../core/theme/app_colors.dart';
import '../../core/theme/app_typography.dart';
import '../../core/theme/app_dimensions.dart';

enum ButtonType {
  primary,
  secondary,
  accent,
}

class CustomButton extends StatelessWidget {
  final String text;
  final VoidCallback? onPressed;
  final ButtonType type;
  final Widget? icon;
  final bool isLoading;
  final double? width;
  final double height;
  final bool disabled;

  const CustomButton({
    super.key,
    required this.text,
    this.onPressed,
    this.type = ButtonType.primary,
    this.icon,
    this.isLoading = false,
    this.width,
    this.height = AppDimensions.buttonHeightLarge,
    this.disabled = false,
  });

  Color get _backgroundColor {
    if (disabled) return AppColors.mediumGray;

    switch (type) {
      case ButtonType.primary:
        return AppColors.primaryGreen;
      case ButtonType.secondary:
        return AppColors.lightGray;
      case ButtonType.accent:
        return AppColors.accentOrange;
    }
  }

  Color get _textColor {
    switch (type) {
      case ButtonType.primary:
        return AppColors.white;
      case ButtonType.secondary:
        return AppColors.darkNavy;
      case ButtonType.accent:
        return AppColors.white;
    }
  }

  @override
  Widget build(BuildContext context) {
    return SizedBox(
      width: width,
      height: height,
      child: ElevatedButton(
        onPressed: disabled || isLoading ? null : onPressed,
        style: ElevatedButton.styleFrom(
          backgroundColor: _backgroundColor,
          foregroundColor: _textColor,
          elevation: AppDimensions.cardElevationLow,
          shadowColor: AppColors.shadowColor,
          shape: RoundedRectangleBorder(
            borderRadius:
                BorderRadius.circular(AppDimensions.borderRadiusMedium),
          ),
          padding: const EdgeInsets.symmetric(
            horizontal: 24,
            vertical: 16,
          ),
        ),
        child: isLoading
            ? SizedBox(
                width: 20,
                height: 20,
                child: CircularProgressIndicator(
                  strokeWidth: 2,
                  valueColor: AlwaysStoppedAnimation<Color>(_textColor),
                ),
              )
            : Row(
                mainAxisSize: MainAxisSize.min,
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  if (icon != null) ...[
                    icon!,
                    const SizedBox(width: 8),
                  ],
                  Text(
                    text,
                    style: AppTypography.labelMedium.copyWith(
                      color: _textColor,
                      fontWeight: FontWeight.w600,
                    ),
                  ),
                ],
              ),
      ),
    );
  }
}

File: src/components/shared/CustomButton.js
import React from 'react';
import {
  TouchableOpacity,
  Text,
  View,
  ActivityIndicator,
  StyleSheet,
} from 'react-native';
import { AppColors, Typography, BorderRadius, ButtonHeight } from '../../theme';

export const ButtonType = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  ACCENT: 'accent',
};

const CustomButton = ({
  text,
  onPress,
  type = ButtonType.PRIMARY,
  icon: Icon,
  isLoading = false,
  width,
  height = ButtonHeight.large,
  disabled = false,
  style,
  textStyle,
}) => {
  const getButtonColors = () => {
    switch (type) {
      case ButtonType.PRIMARY:
        return {
          backgroundColor: AppColors.primaryGreen,
          textColor: AppColors.white,
        };
      case ButtonType.SECONDARY:
        return {
          backgroundColor: AppColors.lightGray,
          textColor: AppColors.darkNavy,
        };
      case ButtonType.ACCENT:
        return {
          backgroundColor: AppColors.accentOrange,
          textColor: AppColors.white,
        };
      default:
        return {
          backgroundColor: AppColors.primaryGreen,
          textColor: AppColors.white,
        };
    }
  };

  const { backgroundColor, textColor } = getButtonColors();

  const buttonStyles = [
    styles.button,
    {
      backgroundColor: disabled ? AppColors.mediumGray : backgroundColor,
      width,
      height,
    },
    style,
  ];

  const textStyles = [
    styles.text,
    { color: textColor },
    textStyle,
  ];

  return (
    <TouchableOpacity
      style={buttonStyles}
      onPress={onPress}
      disabled={disabled || isLoading}
      activeOpacity={0.8}
    >
      {isLoading ? (
        <ActivityIndicator size="small" color={textColor} />
      ) : (
        <View style={styles.content}>
          {Icon && (
            <View style={styles.iconContainer}>
              <Icon size={18} color={textColor} />
            </View>
          )}
          <Text style={textStyles}>{text}</Text>
        </View>
      )}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    borderRadius: BorderRadius.medium,
    paddingHorizontal: 24,
    paddingVertical: 16,
    alignItems: 'center',
    justifyContent: 'center',
    elevation: 2,
    shadowColor: AppColors.shadowColor,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.1,
    shadowRadius: 4,
  },
  content: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  iconContainer: {
    marginRight: 8,
  },
  text: {
    ...Typography.labelMedium,
    fontWeight: '600',
  },
});

export default CustomButton; 